instrumentation-genai/opentelemetry-instrumentation-google-genai/src/opentelemetry/instrumentation/google_genai/generate_content.py

```diff
@@ -505,31 +508,29 @@ def _maybe_log_completion_details(
     def _maybe_log_system_instruction(
         self, config: Optional[GenerateContentConfigOrDict] = None
     ):
-        system_instruction = None
-        if config is not None:
-            if isinstance(config, dict):
-                system_instruction = config.get("system_instruction")
-            else:
-                system_instruction = config.system_instruction
+        content_union = _config_to_system_instruction(config)
+        if not content_union:
+            return
+        content = transformers.t_contents(content_union)[0]
+        if not content.parts:
+            return
+        # System instruction is required to be text. An error will be returned by the API if it isn't.
+        system_instruction = " ".join(
+            part.text for part in content.parts if part.text
+        )
```

Comment:
Could we just model_dump the full provided system instructions Content object without converting into a string? It's true it should be text, but a user might pass non-text and need to debug.